kolibri/core/assets/src/views/StorageNotification.vue

```diff
@@ -47,53 +47,62 @@
 <script>
 
   import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
-  import useUser from 'kolibri.coreVue.composables.useUser';
   import { mapGetters } from 'vuex';
+  import { LearnerDeviceStatus } from 'kolibri.coreVue.vuex.constants';
+  import useUserSyncStatus from '../composables/useUserSyncStatus';
+  import plugin_data from 'plugin_data';
 
   export default {
     name: 'StorageNotification',
     components: {},
     mixins: [commonCoreStrings],
     setup() {
-      const { isLearnerOnlyImport } = useUser();
+      const {
+        status,
+        lastSynced,
+        deviceStatus,
+        deviceStatusSentiment,
+        hasDownloads,
+        lastDownloadRemoved,
+      } = useUserSyncStatus();
+
       return {
-        isLearnerOnlyImport,
+        lastSynced,
+        status,
+        deviceStatus,
+        deviceStatusSentiment,
+        hasDownloads,
+        lastDownloadRemoved,
       };
     },
-    props: {
-      showBanner: {
-        type: Boolean,
-        default: false,
-      },
-    },
     data() {
       return {
-        bannerOpened: false,
-        // TODO: remove this
-        insufficientSpace: true,
-        // TODO: retrieve proper info for these
-        hasDevicePermissions: false,
-        availableDownloads: false,
-        resourcesRemoved: false,
+        isSubsetOfUsersDevice: plugin_data.isSubsetOfUsersDevice,
       };
     },
     computed: {
-      ...mapGetters(['isLearner', 'isAdmin']),
+      ...mapGetters(['isLearner', 'isAdmin', 'canManageContent']),
       insufficientStorageNoDownloads() {
         return (
           (this.isLearner && this.insufficientSpace) ||
-          (!this.hasDevicePermissions && !this.availableDownloads)
+          (!this.canManageContent && !this.hasDownloads)
         );
       },
       learnOnlyRemovedResources() {
-        return this.isLearner && this.resourcesRemoved && this.isLearnerOnlyImport;
+        return this.isLearner && this.lastDownloadRemoved && this.isSubsetOfUsersDevice;
       },
       availableDownload() {
-        return !this.hasDevicePermissions && this.availableDownloads && !this.isLearner;
+        return !this.canManageContent && this.hasDownloads && !this.isLearner;
+      },
+      showBanner() {
+        return (
+          (this.deviceStatus === LearnerDeviceStatus.INSUFFICIENT_STORAGE &&
+            localStorage.getItem('last_synced') < this.lastSynced) ||
```

Comment:
@bjester had suggested using this https://vueuse.org/core/useLocalStorage/ in order to make sure that these localStorage references are reactive, so that this computed prop will be recalculated.